### PR TITLE
rgw: change cls rgw reshard status to enum class

### PR DESCRIPTION
**Pull request by ivancich**
Get the type safety and reduced impact on global namespace of enum class

Tracker: https://tracker.ceph.com/issues/42074

### CODE REVIEW

File: src/rgw/rgw_admin.cc [hunk 1]
```diff
@@ -2575,6 +2575,13 @@ int check_reshard_bucket_params(rgw::sal::RGWRadosStore *store,
     return ret;
   }
 
+  if (bucket_info.reshard_status != cls_rgw_reshard_status::NOT_RESHARDING) {
```
Comment (tchaikov) [inline src/rgw/rgw_admin.cc]:
```
/home/jenkins-build/build/workspace/ceph-pull-requests/src/rgw/rgw_admin.cc: In function 'int check_reshard_bucket_params(rgw::sal::RGWRadosStore*, const string&, const string&, const string&, bool, int, int, rgw_bucket&, RGWBucketInfo&, std::map<std::__cxx11::basic_string<char>, ceph::buffer::v14_2_0::list>&)':
/home/jenkins-build/build/workspace/ceph-pull-requests/src/rgw/rgw_admin.cc:2578:61: error: 'NOT_RESHARDING' is not a member of 'cls_rgw_reshard_status'
   if (bucket_info.reshard_status != cls_rgw_reshard_status::NOT_RESHARDING) {
                                                             ^~~~~~~~~~~~~~
src/rgw/CMakeFiles/radosgw-admin.dir/build.make:62: recipe for target 'src/rgw/CMakeFiles/radosgw-admin.dir/rgw_admin.cc.o' failed
```

can we `#include "cls/rgw/cls_rgw_types.h"` in this source file as well?

Comment (ivancich) [reply src/rgw/rgw_admin.cc]: I did this in an other PR -- https://github.com/ceph/ceph/pull/30610, and it is now in master.

**Comment by ivancich** [comment]
> looks good! just adding a negative review to prevent merging before #30610

Seems rather heavy-handed, @cbodley..... ;)

#30610 has merged. I rebased and am making sure it builds. Once that's verified I'll update the labels and re-request your review.

File: src/cls/rgw/cls_rgw_types.h [hunk 2]
```diff
@@ -638,32 +638,32 @@ struct rgw_bucket_category_stats {
...
     break;
-  default:
-    break;
   };
   return "Unknown reshard status";
 }
 
 struct cls_rgw_bucket_instance_entry {
-  cls_rgw_reshard_status reshard_status{CLS_RGW_RESHARD_NOT_RESHARDING};
+  using RESHARD_STATUS = cls_rgw_reshard_status;
```
Comment (tchaikov) [inline src/cls/rgw/cls_rgw_types.h]: nit, i'd suggest 
```c++
using reshard_status_t = cls_rgw_reshard_status;
```

**Comment by cbodley** [comment]
jenkins test make check

**Comment by ivancich** [comment]
refesh

**Comment by ivancich** [comment]
QA Teuthology Results
verify: http://pulpito.ceph.com/ivancich-2020-02-12_18:05:31-rgw-wip-eric-testing-1-distro-basic-smithi/
non-verify: http://pulpito.ceph.com/ivancich-2020-02-12_18:05:52-rgw-wip-eric-testing-1-distro-basic-smithi/

**Pull request merged**